fastscanner/benchmarks/benchmark_redis_read.py

```diff
@@ -66,24 +66,51 @@ async def monitor_batch_timeout():
         await asyncio.sleep(1)
         now = time.time()
 
-        if (
-            batch_start_time
-            and last_received_time
-            and (now - last_received_time) > NO_DATA_TIMEOUT
-        ):
+        if last_received_time and (now - last_received_time) > NO_DATA_TIMEOUT:
             logger.info(
-                f"\nBatch started at: {datetime.fromtimestamp(batch_start_time).strftime('%M:%S.%f')}"
+                f"No messages received for {NO_DATA_TIMEOUT} seconds at: {datetime.fromtimestamp(now).strftime('%M:%S.%f')[:-3]}"
             )
-            logger.info(
-                f"Batch ended at:   {datetime.fromtimestamp(now).strftime('%M:%S.%f')}"
-            )
-            logger.info(f"Total messages read: {total_messages}")
-            logger.info(f"Batch duration: {now - batch_start_time:.6f} seconds\n")
+            if batch_start_time:
```

Comment:
No need to have two levels of nesting. Also, no need to log when there are no messages for 10 seconds. It's expected because messages arrive every 1 minute. The condition you had before was fine.